TestMonitorErrorLogs is flaky

```
travis_fold:start:worker_info
[0K[33;1mWorker information[0m
hostname: 082c375b-5890-4eb8-b68e-4552678304a8@1.worker-com-75b8f7d6f5-kqrt4.gce-production-1
version: v6.2.17 https://github.com/travis-ci/worker/tree/bc420764f334fe0d096a417c80426f672f1b7d18
instance: travis-job-1138649a-f457-4e6b-bcc4-42539cf7efa3 travis-ci-windows-1809-containers-1592789344-0a98f2fd (via amqp)
startup: 5.89074499s
travis_fold:end:worker_info
[0KSecret environment variables are not obfuscated on Windows, please refer to our documentation: https://docs.travis-ci.com/user/best-practices-security[0m
travis_time:start:0f5a1b00
[0Ktravis_time:end:0f5a1b00:start=1595492997400378400,finish=1595492997440139300,duration=39760900,event=check_unsupported
[0Ktravis_time:start:0594ccc6
[0Ktravis_time:end:0594ccc6:start=1595492997490559300,finish=1595492997530753500,duration=40194200,event=apt_get_update
[0Ktravis_time:start:167a8500
[0K[33;1mDisabling Windows Defender[0m
$ powershell -Command Set-MpPreference -DisableArchiveScanning \$true
$ powershell -Command Set-MpPreference -DisableRealtimeMonitoring \$true
$ powershell -Command Set-MpPreference -DisableBehaviorMonitoring \$true
travis_time:end:167a8500:start=1595492997584002400,finish=1595493000401015400,duration=2817013000,event=disable_windows_defender
[0KUpdating gimme
travis_time:start:0a71287a
[0Ktravis_time:start:089630d0
[0Ktravis_time:end:089630d0:start=1595493021446834300,finish=1595493021634735900,duration=187901600,event=checkout
[0K
travis_fold:start:git.checkout
[0Ktravis_time:start:3cc7aacc
[0K$ git clone --depth=50 https://github.com/GoogleContainerTools/skaffold.git GoogleContainerTools/skaffold
Cloning into 'GoogleContainerTools/skaffold'...
Updating files:  14% (1038/7407)
Updating files:  15% (1112/7407)
Updating files:  16% (1186/7407)
Updating files:  17% (1260/7407)
Updating files:  18% (1334/7407)
Updating files:  19% (1408/7407)
Updating files:  20% (1482/7407)
Updating files:  21% (1556/7407)
Updating files:  22% (1630/7407)
Updating files:  23% (1704/7407)
Updating files:  24% (1778/7407)
Updating files:  25% (1852/7407)
Updating files:  26% (1926/7407)
Updating files:  27% (2000/7407)
Updating files:  28% (2074/7407)
Updating files:  29% (2149/7407)
Updating files:  30% (2223/7407)
Updating files:  31% (2297/7407)
Updating files:  32% (2371/7407)
Updating files:  33% (2445/7407)
Updating files:  34% (2519/7407)
Updating files:  35% (2593/7407)
Updating files:  36% (2667/7407)
Updating files:  36% (2714/7407)
Updating files:  37% (2741/7407)
Updating files:  38% (2815/7407)
Updating files:  39% (2889/7407)
Updating files:  40% (2963/7407)
Updating files:  41% (3037/7407)
Updating files:  42% (3111/7407)
Updating files:  43% (3186/7407)
Updating files:  44% (3260/7407)
Updating files:  45% (3334/7407)
Updating files:  46% (3408/7407)
Updating files:  47% (3482/7407)
Updating files:  48% (3556/7407)
Updating files:  49% (3630/7407)
Updating files:  50% (3704/7407)
Updating files:  51% (3778/7407)
Updating files:  52% (3852/7407)
Updating files:  53% (3926/7407)
Updating files:  54% (4000/7407)
Updating files:  55% (4074/7407)
Updating files:  56% (4148/7407)
Updating files:  57% (4222/7407)
travis_time:end:3cc7aacc:start=1595493021690679900,finish=1595493031386643600,duration=9695963700,event=checkout
[0K$ cd GoogleContainerTools/skaffold
travis_time:start:03adb5ed
[0K$ git fetch origin +refs/pull/4509/merge:
Updating files:  57% (4274/7407)
Updating files:  58% (4297/7407)
Updating files:  59% (4371/7407)
Updating files:  60% (4445/7407)
Updating files:  61% (4519/7407)
Updating files:  62% (4593/7407)
Updating files:  63% (4667/7407)
Updating files:  64% (4741/7407)
Updating files:  65% (4815/7407)
Updating files:  66% (4889/7407)
Updating files:  67% (4963/7407)
Updating files:  68% (5037/7407)
Updating files:  69% (5111/7407)
Updating files:  70% (5185/7407)
Updating files:  71% (5259/7407)
Updating files:  72% (5334/7407)
Updating files:  72% (5346/7407)
Updating files:  73% (5408/7407)
Updating files:  74% (5482/7407)
Updating files:  75% (5556/7407)
Updating files:  76% (5630/7407)
Updating files:  77% (5704/7407)
Updating files:  78% (5778/7407)
Updating files:  79% (5852/7407)
Updating files:  80% (5926/7407)
Updating files:  81% (6000/7407)
Updating files:  82% (6074/7407)
Updating files:  83% (6148/7407)
Updating files:  84% (6222/7407)
Updating files:  85% (6296/7407)
Updating files:  86% (6371/7407)
Updating files:  87% (6445/7407)
Updating files:  88% (6519/7407)
Updating files:  89% (6593/7407)
Updating files:  90% (6667/7407)
Updating files:  91% (6741/7407)
Updating files:  92% (6815/7407)
Updating files:  93% (6889/7407)
Updating files:  94% (6963/7407)
Updating files:  94% (7021/7407)
Updating files:  95% (7037/7407)
Updating files:  96% (7111/7407)
Updating files:  97% (7185/7407)
Updating files:  98% (7259/7407)
Updating files:  99% (7333/7407)
Updating files: 100% (7407/7407)
Updating files: 100% (7407/7407), done.
travis_time:end:03adb5ed:start=1595493031442213400,finish=1595493032149716200,duration=707502800,event=checkout
[0K$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout
[0K
travis_time:end:03adb5ed:start=1595493031442213400,finish=1595493032556317600,duration=1114104200,event=checkout
[0K$ travis_export_go 1.14.x /skaffold
travis_time:start:00037744
[0Ktravis_time:end:00037744:start=1595493032659905900,finish=1595493032784399900,duration=124494000,event=env
[0Ktravis_time:start:017d3f0f
[0K$ travis_setup_go
From https://github.com/GoogleContainerTools/skaffold
 * branch            refs/pull/4509/merge -> FETCH_HEAD
$ export GOPATH="/c/Users/travis/gopath"
$ export PATH="/c/Users/travis/gopath/bin:/c/Users/travis/.gimme/versions/go1.14.6.windows.amd64/bin:/c/Users/travis/bin:/bin:/usr/bin:/c/tools/ruby27/bin:/c/Windows/system32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0:/c/Windows/System32/OpenSSH:/c/ProgramData/GooGet:/c/Program Files/Google/Compute Engine/metadata_scripts:/c/Program Files (x86)/Google/Cloud SDK/google-cloud-sdk/bin:/c/Program Files/Google/Compute Engine/sysprep:/c/Program Files/Docker:/c/ProgramData/chocolatey/bin:/c/Program Files/CMake/bin:/c/Program Files/Git/cmd:/c/Program Files/LLVM/bin:/c/Program Files/dotnet:/c/Program Files (x86)/Windows Kits/10/Windows Performance Toolkit:/c/tools/BCURRAN3:/c/Users/travis/AppData/Local/Microsoft/WindowsApps:/c/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/bin"
$ export GO111MODULE="auto"
go version go1.14.6 windows/amd64
travis_time:end:017d3f0f:start=1595493032840763200,finish=1595493068430741000,duration=35589977800,event=
[0Ktravis_fold:start:cache.1
[0KSetting up build cache
$ export CASHER_DIR=${TRAVIS_HOME}/.casher
travis_time:start:0b9a0e88
[0K$ Installing caching utilities
travis_time:end:0b9a0e88:start=1595493068544007600,finish=1595493068641231500,duration=97223900,event=setup_casher
[0Ktravis_time:start:016da30c
[0Kattempting to download cache archive[0m
[32;1mfetching PR.4509/cache--windows-1809-containers-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--go-1.14.x.tgz[0m
[32;1mfound cache[0m
travis_time:end:016da30c:start=1595493068712460700,finish=1595493070252768200,duration=1540307500,event=setup_casher
[0Ktravis_time:start:070cebcc
[0Kadding C:/Users/travis/AppData/Local/go-build to cache[0m
travis_time:end:070cebcc:start=1595493070304007400,finish=1595493086338414500,duration=16034407100,event=setup_casher
[0Ktravis_fold:end:cache.1
[0K
$ gimme version
v1.5.4
$ go version
go version go1.14.6 windows/amd64
travis_fold:start:go.env
[0K$ go env
set GO111MODULE=auto
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\travis\AppData\Local\go-build
set GOENV=C:\Users\travis\AppData\Roaming\go\env
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=C:/Users/travis/gopath
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=C:\Users\travis\.gimme\versions\go1.14.6.windows.amd64
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=C:\Users\travis\.gimme\versions\go1.14.6.windows.amd64\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=C:\Users\travis\gopath\src\skaffold\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\travis\AppData\Local\Temp\go-build955850094=/tmp/go-build -gno-record-gcc-switches
travis_fold:end:go.env
[0Ktravis_fold:start:install
[0Ktravis_time:start:1a3dd000
[0K$ travis_install_go_dependencies 1.14.x -v
Makefile detected
travis_time:end:1a3dd000:start=1595493087594845100,finish=1595493087637974900,duration=43129800,event=install
[0Ktravis_fold:end:install
[0Ktravis_time:start:0021e2dd
[0K$ go test -short -timeout 60s ./...
?   	github.com/GoogleContainerTools/skaffold/cmd/skaffold	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app	0.401s
ok  	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd	0.809s
ok  	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/config	0.371s
?   	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/credits	[no test files]
?   	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/credits/statik	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/schema	0.251s
?   	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/statik	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/flags	0.288s
?   	github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/tips	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/bazel	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/compose	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/gcb-kaniko	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/generate-pipeline	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/getting-started	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/getting-started-kustomize/app	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/google-cloud-build	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/helm-deployment	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/kaniko	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/microservices/leeroy-app	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/microservices/leeroy-web	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/profile-patches/base-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/profile-patches/hello-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/profile-patches/world-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/profiles/hello-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/profiles/world-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/structure-tests	[no test files]
?   	github.com/GoogleContainerTools/skaffold/examples/tagging-with-environment-variables	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/hack/man	0.608s
ok  	github.com/GoogleContainerTools/skaffold/hack/schemas	0.863s
?   	github.com/GoogleContainerTools/skaffold/hack/tests	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/latest	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/latest_released	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/mark_latest_released	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/new	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/schema_check	[no test files]
?   	github.com/GoogleContainerTools/skaffold/hack/versions/cmd/update_comments	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/hack/versions/pkg/diff	0.277s
ok  	github.com/GoogleContainerTools/skaffold/hack/versions/pkg/schema	0.240s
ok  	github.com/GoogleContainerTools/skaffold/integration	0.780s
ok  	github.com/GoogleContainerTools/skaffold/integration/binpack	(cached)
?   	github.com/GoogleContainerTools/skaffold/integration/examples/bazel	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/compose	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/gcb-kaniko	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/generate-pipeline	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/getting-started	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/getting-started-kustomize/app	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/google-cloud-build	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/helm-deployment	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/kaniko	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/microservices/leeroy-app	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/microservices/leeroy-web	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/profile-patches/base-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/profile-patches/hello-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/profile-patches/world-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/profiles/hello-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/profiles/world-service	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/structure-tests	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/examples/tagging-with-environment-variables	[no test files]
?   	github.com/GoogleContainerTools/skaffold/integration/skaffold	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/diag	0.358s
ok  	github.com/GoogleContainerTools/skaffold/pkg/diag/download	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/diag/recommender	0.352s
ok  	github.com/GoogleContainerTools/skaffold/pkg/diag/validator	0.321s
ok  	github.com/GoogleContainerTools/skaffold/pkg/diag/version	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/apiversion	0.172s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build	1.234s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/bazel	0.370s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/buildpacks	0.684s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache	0.533s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cluster	0.258s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/custom	0.437s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/gcb	0.423s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/jib	0.816s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/list	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/local	0.519s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/misc	0.284s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/tag	0.963s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/color	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/config	0.208s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug	0.776s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy	3.473s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kubectl	0.260s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/resource	0.282s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/diagnose	2.152s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker	10.275s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/errors	0.274s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/event	0.689s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/filemon	0.143s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/gcp	0.228s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/generate_pipeline	0.242s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer	0.751s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/analyze	0.473s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/build	0.292s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/config	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/deploy	0.414s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/errors	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/prompt	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubectl	0.655s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes	0.295s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/context	0.489s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/debugging	0.318s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/generator	0.153s
--- FAIL: TestMonitorErrorLogs (0.64s)
    --- FAIL: TestMonitorErrorLogs/match_on_'error_forwarding_port' (0.35s)
        kubectl_forwarder_test.go:167: cmd was not killed but expected to be killed
FAIL
FAIL	github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/portforward	1.591s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/pipeline	0.230s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner	0.707s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext	0.162s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/util	0.622s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema	1.292s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/defaults	0.552s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/util	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha1	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha2	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha3	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha4	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1alpha5	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta1	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta10	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta11	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta12	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta13	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta14	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta15	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta16	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta17	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta2	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta3	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta4	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta5	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta6	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta7	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta8	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v1beta9	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2alpha1	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2alpha2	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2alpha3	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2alpha4	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2beta1	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2beta2	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2beta3	0.261s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2beta4	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/v2beta5	(cached)
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/validation	0.350s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/server	(cached)
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/sources	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/survey	0.746s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/sync	0.328s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/test	0.383s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/test/structure	0.310s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/trigger	0.309s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/update	0.184s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/util	0.393s
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/version	0.780s
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/walk	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/warnings	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/skaffold/yaml	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/skaffold/yamltags	0.610s
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/constants	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/gcs	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/github	[no test files]
?   	github.com/GoogleContainerTools/skaffold/pkg/webhook/kubernetes	[no test files]
ok  	github.com/GoogleContainerTools/skaffold/pkg/webhook/labels	0.156s
?   	github.com/GoogleContainerTools/skaffold/proto	[no test files]
?   	github.com/GoogleContainerTools/skaffold/testutil	[no test files]
?   	github.com/GoogleContainerTools/skaffold/webhook	[no test files]
FAIL
travis_time:end:0021e2dd:start=1595493087690529400,finish=1595493357257497500,duration=269566968100,event=script
[0K[31;1mThe command "go test -short -timeout 60s ./..." exited with 1.[0m
travis_fold:start:cache.2
[0Kstore build cache
travis_time:start:260b0aa0
[0K[32;1mchanges detected (content changed, file is created, or file is deleted):\nC:\Users\travis\AppData\Local\go-build\00\00050550e591e084a22bab319ba715a0106dfe8d943c43175572f7be8e1a5e9b-d
C:\Users\travis\AppData\Local\go-build\00\0005b08df36cd7a7ab0d13de51c9ab005e1d892c669765d63429bbd9270a2f55-a
C:\Users\travis\AppData\Local\go-build\00\00270df5a31a642c585ef57ef98c5ebb23e339a841502e65f50b58b0539d8527-d
C:\Users\travis\AppData\Local\go-build\00\00db0dd87ec8e2579c0f7853cffe101c3e7c9feeb1bcc5b05b7a997bfdc9a7f5-a
C:\Users\travis\AppData\Local\go-build\00\00e86609e12e2ed48acf742758555bf067a7a8db3911abbb3c58b630ad846450-a
C:\Users\travis\AppData\Local\go-build\00\00f26590984b95a510436924283952856e616e3fde180ff2699ed72af17f42b1-a
C:\Users\travis\AppData\Local\go-build\00\00f7cc3098e6daf43725a9229bd769194bd77b2aaf05916452f1be2c22302f2b-d
C:\Users\travis\AppData\Local\go-build\01\0115e209d406ef97a69ba3c734fed65a4a57d66b806e90dde1aa56ad1777559a-a
C:\Users\travis\AppData\Local\go-build\01\011c6949023ebf38104cda4da8d2c43c9fa4da831c3dcbaa2f0dfddb93750838-a
C:\Users\travis\App\n...[0m
[32;1mchanges detected, packing new archive[0m
gzip: warning: GZIP environment variable is deprecated; use an alias or script
[32;1muploading PR.4509/cache--windows-1809-containers-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--go-1.14.x.tgz[0m
[32;1mcache uploaded[0m
travis_time:end:260b0aa0:start=1595493357311800500,finish=1595493377258705900,duration=19946905400,event=cache
[0Ktravis_fold:end:cache.2
[0K

Done. Your build exited with 1.
```

--- a/pkg/skaffold/kubernetes/portforward/kubectl_forwarder.go
+++ b/pkg/skaffold/kubernetes/portforward/kubectl_forwarder.go
@@ -17,6 +17,7 @@ limitations under the License.
 package portforward
 
 import (
+	"bufio"
 	"bytes"
 	"context"
 	"fmt"
@@ -96,8 +97,8 @@ func (k *KubectlForwarder) forward(parentCtx context.Context, pfe *portForwardEn
 		ctx, cancel := context.WithCancel(parentCtx)
 		pfe.cancel = cancel
 
-		buf := &bytes.Buffer{}
-		cmd := portForwardCmd(ctx, k.kubectl, pfe, buf)
+		var buf bytes.Buffer
+		cmd := portForwardCmd(ctx, k.kubectl, pfe, &buf)
 
 		logrus.Debugf("Running command: %s", cmd.Args)
 		if err := cmd.Start(); err != nil {
@@ -112,7 +113,7 @@ func (k *KubectlForwarder) forward(parentCtx context.Context, pfe *portForwardEn
 		}
 
 		//kill kubectl on port forwarding error logs
-		go k.monitorErrorLogs(ctx, buf, cmd, pfe)
+		go k.monitorErrorLogs(ctx, &buf, cmd, pfe)
 		if err := cmd.Wait(); err != nil {
 			if ctx.Err() == context.Canceled {
 				logrus.Debugf("terminated %v due to context cancellation", pfe)
@@ -158,15 +159,17 @@ func (*KubectlForwarder) Terminate(p *portForwardEntry) {
 // Monitor monitors the logs for a kubectl port forward command
 // If it sees an error, it calls back to the EntryManager to
 // retry the entire port forward operation.
-func (*KubectlForwarder) monitorErrorLogs(ctx context.Context, buf *bytes.Buffer, cmd *kubectl.Cmd, p *portForwardEntry) {
+func (*KubectlForwarder) monitorErrorLogs(ctx context.Context, logs io.Reader, cmd *kubectl.Cmd, p *portForwardEntry) {
+	ticker := time.NewTicker(waitErrorLogs)
+	defer ticker.Stop()
+
+	r := bufio.NewReader(logs)
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		default:
-			time.Sleep(waitErrorLogs)
-
-			s, _ := buf.ReadString(byte('\n'))
+		case <-ticker.C:
+			s, _ := r.ReadString('\n')
 			if s == "" {
 				continue
 			}

--- a/pkg/skaffold/kubernetes/portforward/kubectl_forwarder_test.go
+++ b/pkg/skaffold/kubernetes/portforward/kubectl_forwarder_test.go
@@ -20,6 +20,7 @@ import (
 	"bytes"
 	"context"
 	"runtime"
+	"strings"
 	"sync"
 	"testing"
 	"time"
@@ -114,9 +115,10 @@ func TestMonitorErrorLogs(t *testing.T) {
 
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
-			t.Override(&waitErrorLogs, 50*time.Millisecond)
+			t.Override(&waitErrorLogs, 10*time.Millisecond)
+			ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+			defer cancel()
 
-			ctx, cancel := context.WithCancel(context.Background())
 			cmdStr := "sleep"
 			if runtime.GOOS == "windows" {
 				cmdStr = "timeout"
@@ -130,19 +132,14 @@ func TestMonitorErrorLogs(t *testing.T) {
 			wg.Add(1)
 
 			go func() {
-				defer wg.Done()
+				logs := strings.NewReader(test.input)
+
 				k := KubectlForwarder{}
-				logs := bytes.NewBuffer([]byte(test.input))
 				k.monitorErrorLogs(ctx, logs, cmd, &portForwardEntry{})
+
+				wg.Done()
 			}()
 
-			// need to sleep for one second before cancelling the context
-			// because there is a one second sleep in the switch statement
-			// of monitorLogs
-			time.Sleep(50 * time.Millisecond)
-
-			// cancel the context and then wait for monitorErrorLogs to return
-			cancel()
 			wg.Wait()
 
 			// make sure the command is running or killed based on what's expected

